src/McBopomofo.cpp

```diff
@@ -154,6 +155,36 @@ McBopomofoEngine::McBopomofoEngine(fcitx::Instance* instance)
                                              languageModelLoader_);
   state_ = std::make_unique<InputStates::Empty>();
 
+  editUserPhreasesAction_ = std::make_unique<fcitx::SimpleAction>();
+  editUserPhreasesAction_->setShortText(_("Edit User Phrases"));
+  editUserPhreasesAction_->connect<fcitx::SimpleAction::Activated>(
+      [this](fcitx::InputContext*) {
+        auto command = "xdg-open " + languageModelLoader_->userPhrasesPath();
+        system(command.c_str());
+      });
+  instance_->userInterfaceManager().registerAction(
+      "mcbopomofo-user-phrases-edit", editUserPhreasesAction_.get());
+
+  excludedPhreasesAction_ = std::make_unique<fcitx::SimpleAction>();
+  excludedPhreasesAction_->setShortText(_("Edit Excluded Phrases"));
+  excludedPhreasesAction_->connect<fcitx::SimpleAction::Activated>(
+      [this](fcitx::InputContext*) {
+        auto command =
+            "xdg-open " + languageModelLoader_->excludedPhrasesPath();
+        system(command.c_str());
+      });
+  instance_->userInterfaceManager().registerAction(
+      "mcbopomofo-user-excluded-phrases-edit", excludedPhreasesAction_.get());
+
+  reloadUserPhreasesAction_ = std::make_unique<fcitx::SimpleAction>();
+  reloadUserPhreasesAction_->setShortText(_("Reload User Phrases"));
+  reloadUserPhreasesAction_->connect<fcitx::SimpleAction::Activated>(
+      [this](fcitx::InputContext*) {
+        languageModelLoader_->reloadUserModelsIfNeeded();
```

Comment:
Now we have a menu action here, do you think we should remove the call of `reloadUserModelsIfNeeded()` in `::activate()` (line 252)?

If I log in `::activate`, I'll see that `activate` is called more often on Linux than on macOS, perhaps due to how different apps reset their input method contexts. At any rate checking file timestamp should be cheap, but if our idea is that "user phrase reload is a manually-triggered action", then we don't have to do it in `::activate`?

The thing about doing it in `::activate` right now is that this menu item is basically a no-op, because the user phrases will always be reloaded before they get a chance to use the menu item here.

Reply:
Hmm. I think I can remove the menu item, since I don't think users like to manual update every time. :)

Reply:
LGTM 👍